File: routing-soft-ui/src/app/shared/interfaces/plan-readonly-dto.ts
import { TruckReadOnlyDto } from "./truck-readonly-dto";
import { LocationNodeReadonlyDto } from "./location-node-readonly-dto";
import { RouteReadOnlyDto } from "./route-read-only-dto";

export interface PlanReadonlyDto {
  id: number;
  timeslotLength: number;
  createdAt: string;
  routeDtoList: RouteReadOnlyDto[];
  locationNodeReadOnlyDtoList: LocationNodeReadonlyDto[];
  truckReadOnlyDtoList: TruckReadOnlyDto[];
  planStatus: string;
  algorithmSpec: string;
}

File: routing-soft-core/src/main/java/org/routing/software/dtos/PlanReadOnlyDto.java
package org.routing.software.dtos;

import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class PlanReadOnlyDto {

    private Long id;

    private Long timeslotLength;

    private String createdAt;

    List<RouteDto> routeDtoList;

    List<LocationNodeReadOnlyDto> locationNodeReadOnlyDtoList;

    List<TruckReadOnlyDto> truckReadOnlyDtoList;

    private String planStatus;

    private String algorithmSpec;

    private String uuid; // the UUID of whom the truck belongs
}


